src/fwds_rvs_data.jl

```diff
@@ -861,14 +861,28 @@ tangent type. This method must be equivalent to `tangent_type(_typeof(primal))`.
 @foldable tangent_type(::Type{F}, ::Type{NoRData}) where {F<:Array} = F
 
 # Union types. NOTE: Only `Union{Nothing, Array{<:Base.IEEEFloat}, Base.IEEEFloat}` are supported. 
-for T in [Float16, Float32, Float64]
-    @eval @foldable tangent_type(::Type{NoFData}, ::Type{Union{NoRData,$T}}) = Union{
-        NoTangent,tangent_type($T)
-    }
-    for N in 0:5  # Just go up to N=5 until general solution to https://github.com/chalk-lab/Mooncake.jl/pull/620 available
-        @eval @foldable tangent_type(::Type{Union{NoFData,Array{$T,$N}}}, ::Type{NoRData}) = Union{
-            NoTangent,tangent_type(Array{$T,$N})
-        }
+@foldable function tangent_type(
+    ::Type{NoFData}, ::Type{R}
+) where {R<:Union{NoRData,T} where {T<:Base.IEEEFloat}}
```

Comment:
Instead of having explicit `Base.IEEEFloat` throughout the library maybe you could define something like

```julia
const LeafTypes = Base.IEEEFloat
```

But you would need to reference this throughout the library.

I feel it's a bit easier to read than explicitly referencing specific float types. Also if there's ever a need to include some sort of `Float8` it would be really easy to add it in.

Reply:
Yes yes of course. This is a more general comment for the library, not specific to this issue